### PR TITLE
term_enum_ API support

### PR DESCRIPTION
**Pull request by Vi-cs**
Here is my PR for term enum. Thank you for the review

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## 0.3.11
+- Added support for term enum ([#62](https://github.com/isoos/elastic_client/pull/62) by [Vi-cs](https://github.com/Vi-cs))
+
+## 0.3.10
```
Comment (isoos) [inline CHANGELOG.md:4]: You'll need to rebase this and `pubspec.yaml` on HEAD.

File: lib/src/_client.dart [hunk 2]
```diff
@@ -359,6 +359,40 @@ class Client {
...
+      if (caseInsensitive != null) 'case_insensitive': caseInsensitive,
+      if (size != null) 'size': size,
+    };
+    final rs = await _transport
+        .send(Request('POST', path, bodyMap: map));
+    rs.throwIfStatusNotOK(message: 'Failed to retrieve term enum for $field.');
+    final body = rs.bodyAsMap;
+
+    final termsEntries = body['terms'] as List<dynamic>? ?? const [];
+    final termsResults = termsEntries.map((e) => e as String).toList();
```
Comment (isoos) [inline lib/src/_client.dart]: Use instead: `body['terms']?.whereType<String>()?.toList() ?? <String>[]`

File: lib/src/elastic_client_impl.dart [hunk 3]
```diff
@@ -207,3 +207,9 @@ class ClearScrollResult {
   Map<String, dynamic> toMap() =>
       {'succeeded': succeeded, 'numFreed': numFreed};
 }
+
+class TermEnumResult {
+  final List<String> terms;
+
+  TermEnumResult(this.terms);
+}
```
Comment (isoos) [inline lib/src/elastic_client_impl.dart]: Add extra newline

**Review by isoos — CHANGES_REQUESTED** [review]

File: lib/src/elastic_client_impl.dart (hunk 3 above)
Comment (Vi-cs) [reply lib/src/elastic_client_impl.dart]: I am curious, can you please explain why adding a new line at the end of the file?

Comment (isoos) [reply lib/src/elastic_client_impl.dart]: It's just coding style consistency.

File: lib/src/_client.dart [hunk 4]
```diff
@@ -359,6 +359,39 @@ class Client {
     );
   }
 
+  /// Discover terms in the index that match a partial String
+  Future<TermEnumResult> termEnum({
```
Comment (isoos) [inline lib/src/_client.dart]: Nit: Please add the API URL into the method dartdoc:
https://www.elastic.co/guide/en/elasticsearch/reference/8.1/search-terms-enum.html

Comment (isoos) [inline lib/src/_client.dart]: Nit: Let's call this `termsEnum` instead, as that is the official name of the API. Also, the response should be `TermsEnumResult`.

**Review by isoos — COMMENTED** [review]
Thanks, almost there. I think it is better to follow the official name with the plural `terms`.

**Review by isoos — APPROVED** [review]

**Comment by isoos** [comment]
thanks! published.

**Pull request merged**